tsdb/engine/wal/wal.go

```diff
@@ -0,0 +1,1013 @@
+package wal
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"hash/fnv"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang/snappy"
+	"github.com/influxdb/influxdb/tsdb"
+)
+
+const (
+	DefaultSegmentSize         = 2 * 1024 * 1024 // 2MB, default size of a segment file in a partitions
+	DefaultReadySeriesSize     = 32 * 1024       // 32KB, series is ready to flush once it reaches this size
+	DefaultCompactionThreshold = 0.5             // flush and compact a partition once this ratio of keys are over the flush size
+	DefaultMaxSeriesSize       = 2 * 1024 * 1024 // if any key in a WAL partition reaches this size, we'll force compaction of the partition
+
+	// if there have been no writes to a partition in this amount of time, we'll force a full flush
+	DefaultFlushInterval = 30 * time.Minute
```

Comment:
Go style says to start the comment with the name of the identifier: https://golang.org/doc/effective_go.html#commentary